### DISCUSSION TITLE
Adding authentication requriements with ACR

### DISCUSSION THREAD

**Discussion by mffap** [zitadel/zitadel#5083] [Ideas]
Support [acr claim](https://zitadel.com/docs/apis/openidoauth/claims).

Initial considerations

- Controllable per Project/Application
- Should support NIST and custom values
- Step-up authentication could be done though Actions (checking amr vs. acr)

This feature came up in discussions with enterprise customers, having many SSO-application and want to offer setup-up authentication based on application settings.

**Comment by fforootd** [comment]
Interesting that people now see a need in this.

We anticipated it to some extent since we already have[ placeholder here](https://github.com/zitadel/zitadel/blob/main/internal/api/oidc/auth_request_converter.go#L37)

The AMR we use is in accordance to [oauth ](https://datatracker.ietf.org/doc/html/rfc8176) in our [code ](https://github.com/zitadel/zitadel/blob/main/internal/api/oidc/auth_request_converter.go#L20)

I think if we want to keep this as flexible as possible it would be wise to opt for an action where you can define the acr and map it to the expected amr, otherwise it might be tricky since the standards are not alway well aligned
